addons/l10n_es_edi_facturae/models/account_edi_format.py

```diff
@@ -0,0 +1,499 @@
+
+    def _get_move_applicability(self, move):
+        """ Core function for the EDI processing: it first checks whether the EDI format is applicable on a given
+        move, if so, it then returns a dictionary containing the functions to call for this move.
+
+        :return: dict mapping str to function (callable)
+        * post:             function called for edi.documents with state 'to_send' (post flow)
+        * cancel:           function called for edi.documents with state 'to_cancel' (cancel flow)
+        * post_batching:    function returning the batching key for the post flow
+        * cancel_batching:  function returning the batching key for the cancel flow
+        * edi_content:      function called when computing the edi_content for an edi.document
+        """
+        # EXTENDS account_edi
+        self.ensure_one()
+        if self.code != 'es_face':
+            return super()._get_move_applicability(move)
+
+        if move.company_id.country_code == 'ES':
+            return {
+                'post': self._l10n_es_edi_facturae_post_invoices,
+                'edi_content': self._l10n_es_edi_facturae_xml_edi_content,
+            }
+
+    def _needs_web_services(self):
+        # EXTENDS account_edi
+        return False if self.code == 'es_face' else super()._needs_web_services()
+
+    #############################
+    #   EDI SPECIFIC METHODS    #
+    #############################
+
+    def _l10n_es_edi_facturae_xml_edi_content(self, invoice):
+        return self.l10n_es_edi_facturae_export_facturae(invoice).encode()
+
+    def _l10n_es_edi_facturae_post_invoices(self, invoices):
+        # Ensure a certificate is available.
+        certificate = self.env['l10n_es_edi_facturae.certificate'].search([("company_id", 'in', invoices.company_id.ids)])
+        if not certificate:
+            return {inv: {
+                'error': _("Please configure the certificate for Facturae."),
+                'blocking_level': 'error',
+            } for inv in invoices}
+
+        # Generate the XMLs.
+        res = {}
+        for inv in invoices:
+            xml_name = f'{inv.name.replace("/", "_")}_facturae_signed.xml'
+            attachment = self.env['ir.attachment'].create({
+                'name': xml_name,
+                'raw': self.l10n_es_edi_facturae_render_facturae(inv),
+                'res_model': 'account.edi.document',
+                'mimetype': 'application/xml'
+            })
+            res[inv] = {'success': True, 'attachment': attachment}
+        return res
+
+    @api.model
+    def l10n_es_edi_facturae_clean_xml(self, tree_or_str, as_string=True):
+        """
+        Remove all the unwanted whitespaces and end of lines from the xml tree.
+        :param etree.XML or str tree_or_str: the xml tree in etree or str format
+        :return: the cleaned xml tree
+        """
+        str_tree = tree_or_str if isinstance(tree_or_str, str) else etree.tostring(tree_or_str, encoding=str)
```

Comment:
This method doesn't look clean even if `clean` is part of the method signature :laughing: